src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -946,4 +951,29 @@ public void attachTextPublisher(TextPublisher service) {
     subscribe(service.getName(), "publishText");
   }
 
+  public SearchPublisher getSearch() {
+    return (SearchPublisher)getPeer("search");
+  }
+
+  @Override
+  public String setLocale(String code) {
+   this.locale = new Locale(code);
+   return code;
+  }
+
+  @Override
+  public String getLanguage() {
+   return locale.getLanguage();
+  }
+
+  @Override
+  public Locale getLocale() {
+    return locale;
+  }
+
+  @Override
+  public Map<String, Locale> getLocales() {
+   return Locale.getMap("en-US", "fr-FR", "es-ES", "de-DE", "nl-NL", "ru-RU", "hi-IN","it-IT", "fi-FI","pt-PT");

```

Comment:
This isn't right... only by convention do the inmoov chat bots have locales assocated with them...  as far as program ab is concerned. it actually only supports english or japanese.  (based on the currently implemented tokenziers.

Reply:
I'll let you change it .. I've been unwinding all sorts of surprises I've never seen.  
It would be nice to have a strategy of how "the user" will be less burden by changing or aligning all Locales.

Reply:
Leaving this for you to take care of.